examples/vision/bit.py

```diff
@@ -0,0 +1,308 @@
+from tensorflow import keras
+import tensorflow_hub as hub
+from tensorflow.keras.utils import to_categorical
+
+SEEDS = 42
+
+np.random.seed(SEEDS)
+tf.random.set_seed(SEEDS)
+
+"""
+## Download the dataset
+
+We will be using Image-Scene-Classification dataset which comprises of 30 classes.
+"""
+
+"""shell
+wget -q http://data.vision.ee.ethz.ch/ihnatova/camera_scene_detection_train.zip
+unzip -qq camera_scene_detection_train.zip
+rm -rf camera_scene_detection_train.zip
+"""
+
+"""
+## Data Parsing
+"""
+
+image_paths = list(paths.list_images("training"))
+np.random.shuffle(image_paths)
+image_paths[:5]
+
+"""
+## Counting number of images per class
+"""
+
+labels = []
+for image_path in image_paths:
+    label = image_path.split("/")[1]
+    labels.append(label)
+class_count = Counter(labels)
+pprint(class_count)
+
+"""
+Imbalance Dataset
+"""
+
+"""
+## Define hyperparameters
+"""
+
+RESIZE_TO = 260
+CROP_TO = 224
+TRAIN_SPLIT = 0.9
+BATCH_SIZE = 128
+AUTO = tf.data.AUTOTUNE
+NUM_CLASSES = 30
+STEPS_PER_EPOCH = 140
+
+"""
+## Splitting the Dataset
+"""
+
+i = int(len(image_paths) * TRAIN_SPLIT)
+
+train_paths = image_paths[:i]
+train_labels = labels[:i]
+validation_paths = image_paths[i:]
+validation_labels = labels[i:]
+
+print(len(train_paths), len(validation_paths))
+
+"""
+## Encding the Labels
+"""
+
+le = LabelEncoder()
+train_labels_le = le.fit_transform(train_labels)
+validation_labels_le = le.transform(validation_labels)
+print(train_labels_le[:5])
+
+"""
+Since the dataset has class imbalance issue, it's good to supply class weights while training the model.
+"""
+
+trainLabels = to_categorical(train_labels_le)
+classTotals = trainLabels.sum(axis=0)
+classWeight = dict()
+# loop over all classes and calculate the class weight
+for i in range(0, len(classTotals)):
+    classWeight[i] = classTotals.max() / classTotals[i]
+
+"""
+## Set dataset-dependent hyperparameters
+"""
+
+DATASET_SIZE = "\u003C20k examples"  # @param ["<20k examples", "20k-500k examples", ">500k examples"]
+
+if DATASET_SIZE == "<20k examples":
+    SCHEDULE_LENGTH = 500
+    SCHEDULE_BOUNDARIES = [200, 300, 400]
+elif DATASET_SIZE == "20k-500k examples":
+    SCHEDULE_LENGTH = 10000
+    SCHEDULE_BOUNDARIES = [3000, 6000, 9000]
+else:
+    SCHEDULE_LENGTH = 20000
+    SCHEDULE_BOUNDARIES = [6000, 12000, 18000]
+
+"""
+## Convert the data into TensorFlow Dataset objects
+"""
+
+train_ds = tf.data.Dataset.from_tensor_slices((train_paths, train_labels_le))
+val_ds = tf.data.Dataset.from_tensor_slices((validation_paths, validation_labels_le))
+
+"""
+## Preprocessing helper functions
+"""
+
+SCHEDULE_LENGTH = SCHEDULE_LENGTH * 512 / BATCH_SIZE
+
+
+@tf.function
+def preprocess_train(image_path, label):
+    image = tf.io.read_file(image_path)
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.image.resize(image, (RESIZE_TO, RESIZE_TO))
+    image = tf.image.random_crop(image, [CROP_TO, CROP_TO, 3])
+    image = tf.cast(image, tf.float32) / 255.0
```

Comment:
It's already float32 no?

Reply:
Basically, I meant: do we actually need the cast? The output of `resize` should already be float32.